tests/test_content_cafe.py

```diff
@@ -453,3 +462,117 @@ def create_metadata(self, *args, **kwargs):
         eq_(True, isinstance(result, CoverageFailure))
         eq_(identifier, result.obj)
         assert "Oh no!" in result.exception
+
+
+class MockSession(DummyHTTPClient):
+    """Mock the requests Session object used by zeep.transports.Transport."""
+
+    def __init__(self, wsdl):
+        super(MockSession, self).__init__()
+        self.wsdl = wsdl
+        self.headers = {}
+
+    def get(self, address, *args, **kwargs):
+        # No need to explicitly queue up requests for the WSDL URL --
+        # we know they'll happen a lot.
+        if address == ContentCafeSOAPClient.WSDL_URL:
+            return MockRequestsResponse(200, {}, self.wsdl)
+        return super(MockSession, self).do_get(address)
+
+    def post(self, address, data, *args, **kwargs):
+        return super(MockSession, self).do_post(address, data)
+
+
+class TestContentCafeSOAPClient(ContentCafeAPITest):
+
+    def setup(self):
+        wsdl = self.data_file("service.wsdl")
+        self.session = MockSession(wsdl)
```

Comment:
I don't know much about SOAP or wsdl but they are different XML protocols, right? How do they work together in this case?

Reply:
SOAP is an HTTP protocol for making RPC calls. WSDL is an XML file format that defines the RPC calls and the format of the data that's returned. The WSDL file is how we know how to invoke an RPC method called "DemandHistoryItems". We know that once we call that method we'll get back a data structure that we can navigate as "RequestItems.RequestItem.DemandHistoryItems.DemandHistoryItem", and that each DemandHistoryItem has a Year and a Month. This is all stuff that Content Cafe made up and put into the WSDL file.

So a SOAP client is instantiated with a WSDL file, and that controls which functions you can call on the client.